src/AddAuthorizationHeader.php

```diff
@@ -35,9 +35,8 @@ public function __invoke(RequestInterface $request): RequestInterface
 
     private function getAccessToken(): AccessToken
     {
-        $options = $this->getOptions();
         $grantType = $this->getGrantType();
-        return $this->provider->getAccessToken($grantType, $options);
+        return $this->provider->getAccessToken($grantType, $this->config);
```

Comment:
What header was missing here? In this case the code was filtering out the config values not needed for the headers (for example `grant_type`, as it is passed as parameter.

Maybe we could change how the config is structured and add an "options" key so we can pass that without filtering? What do you think?

Reply:
Actually `get_options` is only passing scope. because you set it as a empty array at the first line of the function `$options = [];`. Maybe we should just remove `grant_type` if it is in there and let the user send any header needed.

In my case I needed `client_id` and `client_secret`

Reply:
`client_id` and `client_secret` should be handled by the abstract provider, passing it by constructor (example in the README.md).
However I think that allowing to pass custom headers is a good idea.
As `$config` is a generic config object, if we want to implement this I propose to change it to look something like this:
```
$config = [
    'grant_type' => 'client_credentials',
    'header_options' => [
        'scopes' => 'my-scope',
        'x-request-id' => 'myrequestId',
        'my-custom-header' => 'Custom Heder',
    ],
];
```
So that we could pass directly `$config['header_options']` to the `getAccessToken` method.
Would you agree?

Reply:
Yes, that would be a great solution IMHO.